src/composeappmanager.cc

```diff
@@ -221,38 +225,45 @@ ComposeAppManager::AppsContainer ComposeAppManager::getAppsToUpdate(const Uptane
         !boost::filesystem::exists(cfg_.apps_root / app_name / Docker::ComposeAppEngine::ComposeFile)) {
       // an App that is supposed to be installed has been removed somehow, let's install it again
       apps_to_update.insert(app_pair);
+      apps_and_reasons[app_pair.first] = "missing installation";
       LOG_INFO << app_name << " will be re-installed";
       continue;
     }
 
     LOG_DEBUG << app_name << " performing full status check";
-    if (!app_engine_->isFetched({app_name, app_pair.second}) || !app_engine_->isRunning({app_name, app_pair.second})) {
-      // an App that is supposed to be installed and running is not fully installed or running
+    if (!app_engine_->isFetched({app_name, app_pair.second})) {
+      // an App that is supposed to be installed is not fully installed
       apps_to_update.insert(app_pair);
-      LOG_INFO << app_name << " update will be re-installed or completed";
+      apps_and_reasons[app_pair.first] = "not fetched";
+      LOG_INFO << app_name << " is not fully installed. Will be re-installed or have update completed";
+      continue;
+    }
+    if (!app_engine_->isRunning({app_name, app_pair.second})) {
+      // an App that is supposed to running is not running
+      apps_to_update.insert(app_pair);
+      apps_and_reasons[app_pair.first] = "not running";
+      LOG_INFO << app_name << " is not running. Will be re-installed or have update completed";
```

Comment:
Maybe `is not installed or not running; will be installed and started`.

Reply:
Updated. You are much more familiar with the details of each app installation situation, so I'm just following your lead on these two suggestions.